### ISSUE TITLE
vocab special token - bos_token_id: 0 , overlapping padding index

### ISSUE THREAD

**Issue by yspaik** [seyonechithrananda/bert-loves-chemistry#6]
Looking at the Chemberta vocab's special id, I found that bos_id=0. In pytorch, sentences of variable length are collated and short sentences are generally set to 0 when padding. In my opinion, it would be better to start with a special token such as bos_token from 1, but is there a reason for assigning it to 0?

#### config.json: seyonec/ChemBERTa-zinc-base-v1
{
"architectures": [
"RobertaForMaskedLM"
],
"attention_probs_dropout_prob": 0.1,
"bos_token_id": 0,
"eos_token_id": 2,
"hidden_act": "gelu",
"hidden_dropout_prob": 0.1,
"hidden_size": 768,
"initializer_range": 0.02,
"intermediate_size": 3072,
"layer_norm_eps": 0.00001,
"max_position_embeddings": 514,
"model_type": "roberta",
"num_attention_heads": 12,
"num_hidden_layers": 6,
"output_past": true,
"pad_token_id": 1,
"type_vocab_size": 1,
"vocab_size": 767
}

**Comment by seyonechithrananda** [comment]
Nice catch! For many specific vocab settings, we use RoBERTa params as in this case. May be worth trying the fix you mentioned and seeing how it performs! Will look into this in future pre-training runs.

**Comment by yspaik** [comment]
I fixed pad_token_id(1) as actual tensor initialization using torch.ones() instead of torch.zero() Now the result is fine as expected. Thank you for your mention.
